src/main/java/uk/gov/hmcts/darts/authorisation/util/EmailAddressFromTokenUtil.java

```diff
@@ -0,0 +1,72 @@
+package uk.gov.hmcts.darts.authorisation.util;
+
+import lombok.experimental.UtilityClass;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.jwt.Jwt;
+
+import java.util.List;
+
+import static java.util.Objects.nonNull;
+
+@UtilityClass
+public class EmailAddressFromTokenUtil {
+    private static final String EMAIL = "email";
+    private static final String EMAILS = "emails";
+    private static final String PREFERRED_USERNAME = "preferred_username";
+
+    public String getEmailAddressFromToken() {
+        Jwt jwt = getJwtFromSecurityContext();
+        Object claimFromJwt = getClaimFromJwt(jwt, List.of(EMAILS, PREFERRED_USERNAME, EMAIL));
+        String emailAddressFromObject = getEmailAddressFromObject(claimFromJwt);
+        if (emailAddressFromObject == null) {
+            throw new IllegalStateException("Could not obtain email address from principal");
```

Comment:
Could make this a darts api exception?

Reply:
this is existing functionality that i have moved, i'm not sure i should be changing the logic too much. i see calling methods are expecting the IllegalStateException.